language/resource_types.md

```diff
@@ -0,0 +1,440 @@
+# Puppet Resource Types
+
+This document describes puppet's DSL for defining puppet resource types. Examples are taken from the `puppetlabs-aws` module and core puppet.
```

Comment:
Small, thing, but think we should say "puppet's Ruby DSL for defining resource types" to avoid confusion with the "puppet DSL" a.k.a "the puppet language", and to avoid confusion with "types" (i.e. data types in general. Later in the text I am fine using just 'type' as a term.